src/native/aws_signing.c

```diff
@@ -69,6 +69,10 @@ static void s_cleanup_callback_data(struct s_aws_sign_request_callback_data *cal
     }
 
     if (callback_data->native_request) {
+        struct aws_input_stream *input_stream = aws_http_message_get_body_stream(callback_data->native_request);
+        if (input_stream != NULL) {
+            aws_input_stream_destroy(input_stream);
```

Comment:
Is it ALWAYS the case that the request here owns the input stream?

Reply:
I think so. The native request is created for signing. (callback_data owns the native request) https://github.com/awslabs/aws-crt-java/blob/main/src/native/aws_signing.c#L381-L382.
And the native request owns the body stream https://github.com/awslabs/aws-crt-java/blob/main/src/native/http_request_utils.c#L326-L332